app/models/action.rb

```diff
@@ -3,5 +3,13 @@ class Action < ActiveRecord::Base
   belongs_to :member
 
   has_paper_trail on: [:update, :destroy]
+
+  after_create :update_member_donor_status
```

Comment:
To clarify, `form_data` is not part of the request - the model would *definitely* be the wrong place for this code were that the case. `form_data` is a JSONB field stored in the row for the action that can hold arbitrary fields depending which form fields the campaigner added to the form.